mender-auth/ipc/server.cpp

```diff
@@ -84,6 +72,35 @@ error::Error Caching::Listen(const string &private_key_path, const string &ident
 	return dbus_server_.AdvertiseObject(dbus_obj);
 }
 
+void Caching::FetchJwtTokenHandler(auth_client::APIResponse &resp) {
+	auth_in_progress_ = false;
+
+	forwarder_.Cancel();
+
+	if (resp) {
+		// ":0" port number means pick random port in user range.
+		auto err = forwarder_.AsyncForward("http://127.0.0.1:0", resp.value().server_url);
+		if (err == error::NoError) {
+			Cache(resp.value().token, forwarder_.GetUrl());
```

Comment:
@vpodzime: The cache still contains `cached_server_url_`, but now it's the URL of the internal forwarder. It makes sense programmatically, but I'm a bit undecided if I'm muddying up the semantics, and making it unclear this way. What do you think?

Reply:
I don't know the story behind this all. But intuitively, I'd keep the old behavior in a class and then made a subclass with a proxy forwarder and serving its URL as the server URL for clients. But if we want to have everything in one class, it makes most sense to just cache the local URL here like your code does.

Reply:
I think using two classes here is a bit overkill, since we only need the functionality of one of them. In that case I'll leave it as it is.